internal/controller/osko/mimirrule_controller.go

```diff
@@ -56,8 +61,10 @@ func (r *MimirRuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			log.V(1).Info("PrometheusRule resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
 		} else {
 			log.Error(err, "Failed to get PrometheusRule")
+			return ctrl.Result{}, err
```

Comment:
what do you think, should we return here or not?

Reply:
I guess that we need to return with the error. Otherwise controller will do next steps which could depend on successfully executing this. Same for the others.

Reply:
so I will change line 64 to return `..., err` as well if we assume that for every mimirrule that exists a PrometheusRule has to exist as well.